ibis/backends/base/sql/alchemy/client.py

```diff
@@ -329,3 +330,101 @@ def _sqla_table_to_expr(self, table):
+            to table_name or
+            the TableExpr type which ibis provides with data which needs
+            to be inserted to table_name
+        database : string, optional
+            name of the attached database that the table is located in.
+        overwrite : boolean, default False
+            If True, will replace existing contents of table else not
+
+        Raises
+        -------
+        NotImplementedError
+            Inserting data to a table from a different database is not
+            yet implemented
+
+        ValueError
+            No operation is being performed. Either the obj parameter
+            is not a pandas DataFrame or is not of string data type for
+            sending table name or is not a ibis Table Expression.
+            The given obj is of type type(obj).__name__ .
+
+        """
+
+        if database == self.database_name:
+            # avoid fully qualified name
+            database = None
+
+        if database is not None:
+            raise NotImplementedError(
+                'Inserting data to a table from a different database is not '
+                'yet implemented'
+            )
+
+        params = {}
+        if self.has_attachment:
+            # for database with attachment
+            # see: https://github.com/ibis-project/ibis/issues/1930
+            params['schema'] = self.database_name
+
+        if isinstance(obj, pd.DataFrame):
+            obj.to_sql(
+                table_name,
+                self.con,
+                index=False,
+                if_exists='replace' if overwrite else 'append',
+                **params,
+            )
+        elif isinstance(obj, str) or isinstance(obj, ir.TableExpr):
+            to_table_expr = self.table(table_name)
+            to_table_schema = to_table_expr.schema()
+
+            if overwrite:
+                self.drop_table(table_name, database=database)
+                self.create_table(
+                    table_name, schema=to_table_schema, database=database,
+                )
+
+            to_table = self._get_sqla_table(table_name, schema=database)
+
+            from_table_expr = None
+            if isinstance(obj, str):
+                from_table_expr = self.table(obj)
+            elif isinstance(obj, ir.TableExpr):
+                from_table_expr = obj
+
+            with self.begin() as bind:
+                if from_table_expr is not None:
+                    bind.execute(
+                        to_table.insert().from_select(
+                            list(from_table_expr.columns),
+                            from_table_expr.compile(),
+                        )
+                    )
+        else:
+            raise ValueError(
+                f"No operation is being performed. Either the obj parameter"
+                f" is not a pandas DataFrame or is not of string data type for"
+                f"sending table name or is not a ibis Table Expression."
```

Comment:
No big deal, but the `f` is only needed in the lines where there is a variable, not all them.

Reply:
Fixed!